dp-hello-world-event/event/consumer.go

```diff
@@ -0,0 +1,104 @@
+package event
+
+import (
+	"context"
+	"errors"
+
+	"github.com/ONSdigital/dp-hello-world-event/schema"
+	kafka "github.com/ONSdigital/dp-kafka/v2"
+	"github.com/ONSdigital/log.go/log"
+)
+
+//go:generate moq -out mock/handler.go -pkg mock . Handler
+
+// MessageConsumer provides a generic interface for consuming []byte messages
+type MessageConsumer interface {
+	Channels() *kafka.ConsumerGroupChannels
+}
+
+// Handler represents a handler for processing a single event.
+type Handler interface {
+	Handle(ctx context.Context, helloCalled *HelloCalled) error
+}
+
+// Consumer consumes event messages.
+type Consumer struct {
```

Comment:
I think we might not need this struct anymore if we do the suggested changes in Consume()